src/routes/jira/workspaces/jira-workspaces-get.test.ts

```diff
@@ -0,0 +1,248 @@
+			host: jiraHost,
+			installationId: 1234,
+			hashedClientKey: "key-123",
+			gitHubAppId: undefined
+		});
+
+		repo = {
+			subscriptionId: sub.id,
+			repoId: 1,
+			repoName: "github-for-jira",
+			repoOwner: "atlassian",
+			repoFullName: "atlassian/github-for-jira",
+			repoUrl: "github.com/atlassian/github-for-jira"
+		};
+
+		jwt = encodeSymmetric({
+			qsh: "context-qsh",
+			iss: "jira-client-key"
+		}, await installation.decrypt("encryptedSharedSecret", getLogger("test")));
+	});
+
+	it("Should return a 400 status if no Subscription is found for host", async () => {
+		app = express();
+		app.use((req, _, next) => {
+			req.log = getLogger("test");
+			next();
+		});
+		app.use(getFrontendApp());
+
+		await Subscription.uninstall({
+			installationId: 1234,
+			host: jiraHost,
+			gitHubAppId: undefined
+		});
+
+		await supertest(app)
+			.get("/jira/workspaces/search?searchQuery=Atlas")
+			.query({
+				jwt
+			})
+			.expect(res => {
+				expect(res.status).toBe(400);
+				expect(res.text).toContain(MISSING_SUBSCRIPTION);
+			});
+	});
+
+	it("Should return a 400 status if no org name is provided in query params", async () => {
+		app = express();
+		app.use((req, _, next) => {
+			req.log = getLogger("test");
+			next();
+		});
+		app.use(getFrontendApp());
+
+		await supertest(app)
+			.get("/jira/workspaces/search")
+			.query({
+				jwt
+			})
+			.expect(res => {
+				expect(res.status).toBe(400);
+				expect(res.text).toContain(MISSING_ORG_NAME);
+			});
+	});
+
+	it("Should return a 400 status if no matching repo is found for orgName + subscription id", async () => {
+		app = express();
+		app.use((req, _, next) => {
+			req.log = getLogger("test");
+			next();
+		});
+		app.use(getFrontendApp());
+
+		await supertest(app)
+			.get("/jira/workspaces/search?searchQuery=incorrectorgname")
+			.query({
+				jwt
+			})
+			.expect(res => {
+				expect(res.status).toBe(400);
+				expect(res.text).toContain(NO_MATCHING_WORKSPACES);
+			});
+	});
+
+	it("Should return a single org if only one match is found", async () => {
+		app = express();
+		app.use((req, _, next) => {
+			req.log = getLogger("test");
+			next();
+		});
+		app.use(getFrontendApp());
+
+		await RepoSyncState.create({
+			...repo,
+			subscriptionId: sub.id
+		});
+
+		const workspaces = {
+			success:true,
+			workspaces: [
+				{
+					id: sub.id.toString(),
+					name: "atlassian",
+					canCreateContainer: false
+				}
+			]
+		};
+
+		await supertest(app)
+			.get("/jira/workspaces/search?searchQuery=atlas")
+			.query({
+				jwt
+			})
+			.expect(res => {
+				expect(res.text).toContain(JSON.stringify(workspaces));
+				expect(res.status).toBe(200);
+			});
+	});
+
+	it("Should return multiple matches but only return each org once", async () => {
+		app = express();
+		app.use((req, _, next) => {
+			req.log = getLogger("test");
+			next();
+		});
+		app.use(getFrontendApp());
+
+		const sub2 = await Subscription.install({
```

Comment:
Can you extract some method to condense this test? It will make test more readble.